src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/actiongraph/ActionGraphV1Parser.java

```diff
@@ -67,7 +50,9 @@ private String getTargetId(String needle) {
         .collect(Collectors.toList());
   }
 
-  private List<AnalysisProtos.Artifact> expandDepsetToArtifacts(Queue<String> idsToExpand) {
+  private List<AnalysisProtos.Artifact> expandDepsetToArtifacts(String idToExpand) {
+    Queue<String> idsToExpand = new ArrayDeque<>(Lists.newArrayList(idToExpand));
```

Comment:
well, this queue will contain only one element - it can be replaced with `Optional` and then execute filtering below

Reply:
I don't think it's that simple. Later there is this while that does `idsToExpand.addAll(` to this queue.

Reply:
It looks like some kind of BFS.

Reply:
ohh right, sorry